five/nineteen.py

```diff
@@ -15,10 +15,42 @@ def rotate_90_clock_np(arr):
     return arr_rot.astype(int).tolist()
 
 
+def rotate_coords_90_clock(i, j, n):
+    i_rot = j
+    j_rot = n - 1 - i
+    return i_rot, j_rot
+
+
+def rotate_corners(arr, i_start, j_start):
+    n = len(arr)
+    i_orig, j_orig = i_start, j_start
+    reassign_val = arr[i_orig][j_orig]
+    for _ in range(4):
+        i_rot, j_rot = rotate_coords_90_clock(i_orig, j_orig, n)
+        temp = arr[i_rot][j_rot]
+        arr[i_rot][j_rot] = reassign_val
+        reassign_val = temp
+        i_orig, j_orig = i_rot, j_rot
+
+        
+def rotate_square(arr, ind_start):
+    i_start = ind_start
+    for j_start in range(ind_start, len(arr) - ind_start - 1):
+        rotate_corners(arr, i_start, j_start)
+
+        
+def rotate_90_clock_inplace(arr):
```

Comment:
Instead of this function and the `rotate_square` function you should be able to call `rotate_corners` in a loop like

```python
def rotate(arr):
  for i in xrange(len(arr)/2):
    for j in xrange(len(arr)/2 + len(arr)%2):
      rotate_corners(arr,i,j)
  return arr
```

Reply:
yeah agreed definitely cleaner, although i think you'd want
`for j in xrange(i, len(arr) - i - 1):`
the original way was more of a direct mapping to how i conceptualized the problem

Reply:
ah i see, we're just calculating quadrants of the square differently